### PR TITLE
Improvements to `ThreadSafeBox`

### PR DESCRIPTION
**Pull request by neonichu**
- Offer a mutate function
- Offer an append function for boxed strings

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

File: Sources/Basics/Concurrency/ThreadSafeBox.swift [hunk 1]
```diff
@@ -24,6 +24,13 @@ public final class ThreadSafeBox<Value> {
         self.underlying = seed
     }
 
+    public func mutate(body: (Value?) throws -> Value) rethrows {
```
[the remark this answers is not in the document]
Comment (tomerd) [reply Sources/Basics/Concurrency/ThreadSafeBox.swift:27]: +1 
`public func mutate(body: (Value?) throws -> Value) rethrows -> Value`

File: Sources/Basics/Concurrency/ThreadSafeBox.swift [hunk 2]
```diff
@@ -100,4 +107,12 @@ extension ThreadSafeBox where Value == Int {
     }
 }
 
+extension ThreadSafeBox where Value == String {
+    public func append(_ value: String) {
+        self.mutate { existingValue in
+            return (existingValue ?? "") + value
```
Comment (tomerd) [inline Sources/Basics/Concurrency/ThreadSafeBox.swift:113]:
```
if let existingValue {
  existingValue + value
} else {
  value
}
```

**Review by tomerd — APPROVED** [review]
cool. some ideas

**Review by MaxDesiatov — APPROVED** [review]

File: Sources/Basics/Concurrency/ThreadSafeBox.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (neonichu) [reply Sources/Basics/Concurrency/ThreadSafeBox.swift:27]: What's the behavior for nil? no op or actually nil the underlying value?

[the remark this answers is not in the document]
Comment (yim-lee) [reply Sources/Basics/Concurrency/ThreadSafeBox.swift:27]: i think it should nil the underlying value

[the remark this answers is not in the document]
Comment (tomerd) [reply Sources/Basics/Concurrency/ThreadSafeBox.swift:27]: +1

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Pull request merged**